airflow/models/baseoperator.py

```diff
@@ -1072,6 +1072,8 @@ def __init__(
         if SetupTeardownContext.active:
             SetupTeardownContext.update_context_map(self)
 
+        self._start_trigger: BaseTrigger | None = getattr(self, "_start_trigger", None)
```

Comment:
This patterns seems weird. Why do we try to get an attribue on self and assign the result back to self in `__init__`? When can the attribute be set before this point?

Reply:
```python
        class Test2Operator(BaseOperator):
            def __init__(self, *args, **kwargs):
                self._start_trigger = trigger
                self._next_method = "execute_complete"
                super().__init__(*args, **kwargs)

            def execute_complete(self):
                pass
```

I'm unsure whether users will do something like this or if we should allow them. Just want to prevent user from doing it in a not ideal (or even wrong?) way. Can change it back to `= _start_trigger` if we do not want there user to do something like this

Reply:
Yep, sounds reasonable. Just updated